diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/import/frgcpa_.inc MCNP_CODE/MCNP6/Source_new/import/frgcpa_.inc
--- MCNP_CODE/MCNP6/Source/import/frgcpa_.inc
+++ MCNP_CODE/MCNP6/Source_new/import/frgcpa_.inc
@@ -3 +3 @@
-!$OMP THREADPRIVATE (/FRGCPA_/)
+! !$OMP THREADPRIVATE (/FRGCPA_/)
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/import/frgspa_.inc MCNP_CODE/MCNP6/Source_new/import/frgspa_.inc
--- MCNP_CODE/MCNP6/Source/import/frgspa_.inc
+++ MCNP_CODE/MCNP6/Source_new/import/frgspa_.inc
@@ -3 +3 @@
-!$OMP THREADPRIVATE (/FRGSPA_/)
+! !$OMP THREADPRIVATE (/FRGSPA_/)
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/src/charged_particle_history.F90 MCNP_CODE/MCNP6/Source_new/src/charged_particle_history.F90
--- MCNP_CODE/MCNP6/Source/src/charged_particle_history.F90
+++ MCNP_CODE/MCNP6/Source_new/src/charged_particle_history.F90
@@ -580 +580,5 @@
-          call uplpos(xt,uold,pbl%i%lev,D,v,0)
+          if (am /= one) then
+            call uplpos(xt,uold,pbl%i%lev,D,v,1)
+          else
+            call uplpos(xt,uold,pbl%i%lev,D,v,0)
+          endif
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/src/exemes.F90 MCNP_CODE/MCNP6/Source_new/src/exemes.F90
--- MCNP_CODE/MCNP6/Source/src/exemes.F90
+++ MCNP_CODE/MCNP6/Source_new/src/exemes.F90
@@ -273 +273 @@
-      select case( hm(js) )
+      select case( trim(hm(js)) )
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/src/main.F90 MCNP_CODE/MCNP6/Source_new/src/main.F90
--- MCNP_CODE/MCNP6/Source/src/main.F90
+++ MCNP_CODE/MCNP6/Source_new/src/main.F90
@@ -65 +65 @@
-  use mcnp_env, only : HDPTH0, KOD, LODDAT, VER
+  use mcnp_env, only : hdpth0, kod, loddat, ver
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/src/mcnp_env.F90 MCNP_CODE/MCNP6/Source_new/src/mcnp_env.F90
--- MCNP_CODE/MCNP6/Source/src/mcnp_env.F90
+++ MCNP_CODE/MCNP6/Source_new/src/mcnp_env.F90
@@ -0,0 +1,14 @@
+
+  module mcnp_env
+    !
+    ! code & environment info for mcnp
+    !
+
+    character(len=8), parameter ::  kod            = KODE
+    character(len=5), parameter ::  ver            = VERS
+    character(len=20),parameter ::  thread         = THREAD
+    character(len=20),parameter ::  thread_version = TVERS
+    character(len=8), parameter ::  loddat         = LODDAT
+    character(len=*), parameter ::  hdpth0         = DPATH
+
+  end module mcnp_env
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/src/torus.F90 MCNP_CODE/MCNP6/Source_new/src/torus.F90
--- MCNP_CODE/MCNP6/Source/src/torus.F90
+++ MCNP_CODE/MCNP6/Source_new/src/torus.F90
@@ -90 +90,7 @@
-      if( abs(t3)<=z0 )  go to 40
+      if( abs(t3)<=z0 ) then
+        if (xx(m) < 0.) then
+          cycle DO_60
+        else
+          go to 40
+        endif
+      endif
diff -rN '--unified=0' MCNP_CODE/MCNP6/Source/src/trfsrf.F90 MCNP_CODE/MCNP6/Source_new/src/trfsrf.F90
--- MCNP_CODE/MCNP6/Source/src/trfsrf.F90
+++ MCNP_CODE/MCNP6/Source_new/src/trfsrf.F90
@@ -200 +200 @@
-  scf(l+1:l+3) = scf(l+1:l+3) - trf(2:3,jt)
+  scf(l+1:l+3) = scf(l+1:l+3) - trf(2:4,jt)
