diff -rN '--unified=0' MCNPX/Source/src/mcnpx/mcnpf/GLOBAL1_zc.F MCNPX/Source_new/src/mcnpx/mcnpf/GLOBAL1_zc.F
--- MCNPX/Source/src/mcnpx/mcnpf/GLOBAL1_zc.F
+++ MCNPX/Source_new/src/mcnpx/mcnpf/GLOBAL1_zc.F
@@ -527,9 +527,5 @@
-      character*8, parameter :: kod = 'mcnpx'                                   !/*mgp       486*/
-      character*5, parameter :: ver = '2.7.0'                                   !/*mgp7g       1*/
-      character*28, parameter :: loddat = 'Mon Apr 18 08:00:00 MST 2011'        !/*mgp7g       2*/
-#if defined(DEC) && !defined(UNIX)                                              /*!mgp       489*/
-      character*80, parameter :: hdpth0 = 'c:\mcnpx\xs'                         !/*mgp       490*/
-#else                                                                           /*!mgp       491*/
-!     character*80, parameter :: hdpth0 = LIBPREFIX                             !/*mgpc        8*/
-      character*80, parameter :: hdpth0 = '/usr/local/lib'                      !/*mgpk        4*/
-#endif                                                                          /*!mgp       494*/
+      character*8, parameter :: kod = KODE                                      !/*mgp       486*/
+      character*5, parameter :: ver = VERS                                      !/*mgp7g       1*/
+      character*28, parameter :: loddat = LODDAT                                !/*mgp7g       2*/
+      character*80, parameter :: hdpth0 =
+     &DPATH                                                                     !/*mgpc        8*/
diff -rN '--unified=0' MCNPX/Source/src/mcnpx/mcnpf/laqmod31.F MCNPX/Source_new/src/mcnpx/mcnpf/laqmod31.F
--- MCNPX/Source/src/mcnpx/mcnpf/laqmod31.F
+++ MCNPX/Source_new/src/mcnpx/mcnpf/laqmod31.F
@@ -2435,2 +2435,3 @@
-      data nba/2*0,2*1,2,3,3,4,0,0,1/
-      data ma/2*0.140,2*0.940,1.88,2*2.82,3.76,2*0.492,0.940/
+      data nba/0,0,1,1,2,3,3,4,0,0,1/
+      data ma/0.140,0.140,0.940,0.940,1.88,2.82,2.82,3.76,0.492,0.492,
+     &0.940/
