diff -rN '--unified=0' MCNPX/Source/src/mcnpx/mcnpf/GLOBAL1_zc.F MCNPX/Source_new/src/mcnpx/mcnpf/GLOBAL1_zc.F
--- MCNPX/Source/src/mcnpx/mcnpf/GLOBAL1_zc.F	2017-07-11 02:43:54.785533171 -0500
+++ MCNPX/Source_new/src/mcnpx/mcnpf/GLOBAL1_zc.F	2017-07-11 13:23:35.601882219 -0500
@@ -527,9 +527,4 @@
-      character*8, parameter :: kod = 'mcnpx'                                   !/*mgp       486*/
-      character*5, parameter :: ver = '2.7.0'                                   !/*mgp7g       1*/
-      character*28, parameter :: loddat = 'Mon Apr 18 08:00:00 MST 2011'        !/*mgp7g       2*/
-#if defined(DEC) && !defined(UNIX)                                              /*!mgp       489*/
-      character*80, parameter :: hdpth0 = 'c:\mcnpx\xs'                         !/*mgp       490*/
-#else                                                                           /*!mgp       491*/
-!     character*80, parameter :: hdpth0 = LIBPREFIX                             !/*mgpc        8*/
-      character*80, parameter :: hdpth0 = '/usr/local/lib'                      !/*mgpk        4*/
-#endif                                                                          /*!mgp       494*/
+      character*8, parameter :: kod = KODE                                      !/*mgp       486*/
+      character*5, parameter :: ver = VERS                                      !/*mgp7g       1*/
+      character*28, parameter :: loddat = LODDAT                                !/*mgp7g       2*/
+      character*80, parameter :: hdpth0 = DPATH                                 !/*mgpc        8*/
