diff -rN '--unified=0' MCNP515/Source/dotcomm/include/dotcomm.h MCNP515/Source_new/dotcomm/include/dotcomm.h
--- MCNP515/Source/dotcomm/include/dotcomm.h	2024-05-21 10:51:09.963041249 -0500
+++ MCNP515/Source_new/dotcomm/include/dotcomm.h	2024-05-21 12:30:15.042477364 -0500
@@ -165 +165 @@
-#define DOTCOMM_Datatype   int
+#define DOTCOMM_Datatype   MPI_Datatype
@@ -585 +585 @@
-				 int             *datatype,
+				 DOTCOMM_Datatype *datatype,
diff -rN '--unified=0' MCNP515/Source/dotcomm/src/internals/mpi/dotcommi_enroll.c MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommi_enroll.c
--- MCNP515/Source/dotcomm/src/internals/mpi/dotcommi_enroll.c	2024-05-21 10:51:09.963041249 -0500
+++ MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommi_enroll.c	2024-05-21 12:34:01.573964822 -0500
@@ -60,0 +61,2 @@
+  DOTCOMM_Context
+    default_user_context;   /* Set to MPI_COMM_WORLD.                        */
@@ -62 +63,0 @@
-    default_user_context,   /* Set to MPI_COMM_WORLD.                        */
diff -rN '--unified=0' MCNP515/Source/dotcomm/src/internals/mpi/dotcommi_pack.c MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommi_pack.c
--- MCNP515/Source/dotcomm/src/internals/mpi/dotcommi_pack.c	2024-05-21 10:51:09.963041249 -0500
+++ MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommi_pack.c	2024-05-21 12:28:24.652443758 -0500
@@ -60 +60 @@
-void DOTCOMMI_PACK ( const void *pv, int *count, int *datatype, int *ierr)
+void DOTCOMMI_PACK ( const void *pv, int *count, DOTCOMM_Datatype *datatype, int *ierr)
diff -rN '--unified=0' MCNP515/Source/dotcomm/src/internals/mpi/dotcommpi_mpi.h MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommpi_mpi.h
--- MCNP515/Source/dotcomm/src/internals/mpi/dotcommpi_mpi.h	2024-05-21 10:51:09.963041249 -0500
+++ MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommpi_mpi.h	2024-05-21 12:26:26.902105282 -0500
@@ -35 +35 @@
-  int dotcommpi_sizeof ( int datatype );
+  int dotcommpi_sizeof ( DOTCOMM_Datatype datatype );
diff -rN '--unified=0' MCNP515/Source/dotcomm/src/internals/mpi/dotcommpi_sizeof.c MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommpi_sizeof.c
--- MCNP515/Source/dotcomm/src/internals/mpi/dotcommpi_sizeof.c	2024-05-21 10:51:09.963041249 -0500
+++ MCNP515/Source_new/dotcomm/src/internals/mpi/dotcommpi_sizeof.c	2024-05-21 12:30:22.054918956 -0500
@@ -40 +40 @@
-int dotcommpi_sizeof ( int datatype )
+int dotcommpi_sizeof ( DOTCOMM_Datatype datatype )
diff -rN '--unified=0' MCNP515/Source/src/torus.F90 MCNP515/Source_new/src/torus.F90
--- MCNP515/Source/src/torus.F90
+++ MCNP515/Source_new/src/torus.F90
@@ -68 +68,7 @@
-      if( abs(t3)<=z0 )  go to 40
+      if( abs(t3)<=z0 ) then
+        if (xx(m) < 0.) then
+          cycle DO_60
+        else
+          go to 40
+        endif
+      endif
