diff -rN '--unified=0' MCNP611/Source/import/frgcpa_.inc MCNP611/Source_new/import/frgcpa_.inc
--- MCNP611/Source/import/frgcpa_.inc
+++ MCNP611/Source_new/import/frgcpa_.inc
@@ -3 +3 @@
-!$OMP THREADPRIVATE (/FRGCPA_/)
+! !$OMP THREADPRIVATE (/FRGCPA_/)
diff -rN '--unified=0' MCNP611/Source/import/frgspa_.inc MCNP611/Source_new/import/frgspa_.inc
--- MCNP611/Source/import/frgspa_.inc
+++ MCNP611/Source_new/import/frgspa_.inc
@@ -3 +3 @@
-!$OMP THREADPRIVATE (/FRGSPA_/)
+! !$OMP THREADPRIVATE (/FRGSPA_/)
diff -rN '--unified=0' MCNP611/Source/import/strexc.inc MCNP611/Source_new/import/strexc.inc
--- MCNP611/Source/import/strexc.inc
+++ MCNP611/Source_new/import/strexc.inc
@@ -3 +3 @@
-!$OMP THREADPRIVATE (/STREXC/)
+! !$OMP THREADPRIVATE (/STREXC/)
diff -rN '--unified=0' MCNP611/Source/src/cgm_interface.F90 MCNP611/Source_new/src/cgm_interface.F90
--- MCNP611/Source/src/cgm_interface.F90
+++ MCNP611/Source_new/src/cgm_interface.F90
@@ -29,2 +29,2 @@
-      import :: c_double
-      real (kind=c_double), external, bind(c) :: rngfunc
+      import :: c_funptr
+      type(c_funptr), value :: rngfunc
diff -rN '--unified=0' MCNP611/Source/src/charged_particle_history.F90 MCNP611/Source_new/src/charged_particle_history.F90
--- MCNP611/Source/src/charged_particle_history.F90
+++ MCNP611/Source_new/src/charged_particle_history.F90
@@ -594 +594,5 @@
-          call uplpos(xt,uold,pbl%i%lev,D,v,0)
+          if (am /= one) then
+            call uplpos(xt,uold,pbl%i%lev,D,v,1)
+          else
+            call uplpos(xt,uold,pbl%i%lev,D,v,0)
+          endif
diff -rN '--unified=0' MCNP611/Source/src/exemes.F90 MCNP611/Source_new/src/exemes.F90
--- MCNP611/Source/src/exemes.F90
+++ MCNP611/Source_new/src/exemes.F90
@@ -273 +273 @@
-      select case( hm(js) )
+      select case( trim(hm(js)) )
diff -rN '--unified=0' MCNP611/Source/src/imcn.F90 MCNP611/Source_new/src/imcn.F90
--- MCNP611/Source/src/imcn.F90
+++ MCNP611/Source_new/src/imcn.F90
@@ -36,0 +37 @@
+  use, intrinsic :: iso_c_binding
@@ -2033 +2034 @@
-      call setrngcgm(rang)          
+      call setrngcgm(c_funloc(rang))
@@ -2299 +2300 @@
-      call setrngcgm(rang)          
+      call setrngcgm(c_funloc(rang))
diff -rN '--unified=0' MCNP611/Source/src/main.F90 MCNP611/Source_new/src/main.F90
--- MCNP611/Source/src/main.F90
+++ MCNP611/Source_new/src/main.F90
@@ -64 +64 @@
-  use mcnp_env, only : HDPTH0, KOD, LODDAT, VER
+  use mcnp_env, only : hdpth0, kod, loddat, ver
diff -rN '--unified=0' MCNP611/Source/src/mc.c MCNP611/Source_new/src/mc.c
--- MCNP611/Source/src/mc.c
+++ MCNP611/Source_new/src/mc.c
@@ -485 +485 @@
-  extern void xevent();
+  extern void xevent(IDEF wait);
@@ -517 +517 @@
-  extern void xevent();
+  extern void xevent(IDEF wait);
@@ -576 +576 @@
-  extern void xevent();
+  extern void xevent(IDEF wait);
diff -rN '--unified=0' MCNP611/Source/src/mcnp_env.F90 MCNP611/Source_new/src/mcnp_env.F90
--- MCNP611/Source/src/mcnp_env.F90
+++ MCNP611/Source_new/src/mcnp_env.F90
@@ -0,0 +1,77 @@
+
+  module mcnp_env
+    !
+    ! code & environment info for mcnp
+    !
+
+    character(len=8), parameter ::  kod            = KODE
+    character(len=5), parameter ::  ver            = VERS
+    character(len=20),parameter ::  thread         = THREAD
+    character(len=20),parameter ::  thread_version = TVERS
+    character(len=8), parameter ::  loddat         = LODDAT
+    character(len=*), parameter ::  hdpth0         = DPATH
+
+    public ::  print_build_info
+  CONTAINS
+    subroutine print_build_info( iunit )
+      integer, intent(in) :: iunit
+
+      write(iunit,*) '  +-----------------------------------------------------------'
+      write(iunit,*) '  | mcnp6 build information:'
+      write(iunit,*) '  |'
+      write(iunit,*) '  | This executable was built with a custom CMake build, not'
+      write(iunit,*) '  | the build system from the discs supplied by RSICC. See'
+      write(iunit,*) '  | https://github.com/ljacobson64/MCNP_CMake for more'
+      write(iunit,*) '  | information.'
+      write(iunit,*) '  |'
+      write(iunit,*) '  |   user          = ', ENV_USER
+      write(iunit,*) '  |   host          = ', ENV_HOST
+      write(iunit,*) '  |   OS            = ', ENV_OS
+      write(iunit,*) '  |   date          = ', ENV_DATE
+      write(iunit,*) '  |   time          = ', ENV_TIME
+      write(iunit,*) '  |'
+      write(iunit,*) '  |   cc            = ', ENV_CC
+      write(iunit,*) '  |   cxx           = ', ENV_CXX
+      write(iunit,*) '  |   f90           = ', ENV_F90
+#ifdef OMP
+      write(iunit,*) '  |   OpenMP        = yes'
+#else
+      write(iunit,*) '  |   OpenMP        = no'
+#endif
+#ifdef MPI
+      write(iunit,*) '  |   MPI           = yes'
+      write(iunit,*) '  |   MPI_ROOT      = ', ENV_MPI_ROOT
+#else
+      write(iunit,*) '  |   MPI           = no'
+#endif
+      write(iunit,*) '  |'
+      write(iunit,*) '  |   code          = ', KODE
+      write(iunit,*) '  |   version       = ', VERS
+      write(iunit,*) '  |'
+      write(iunit,*) '  |   thread name   = ', THREAD
+      write(iunit,*) '  |   thread number = ', TVERS
+      write(iunit,*) '  |'
+      write(iunit,*) '  |   datapath      = ', DPATH
+      write(iunit,*) '  |'
+      write(iunit,*) '  |   fpp defs      = '
+      write(iunit,*) '  |                   ', COMPDEF_0
+      write(iunit,*) '  |                   ', COMPDEF_1
+      write(iunit,*) '  |                   ', COMPDEF_2
+      write(iunit,*) '  |                   ', COMPDEF_3
+      write(iunit,*) '  |                   ', COMPDEF_4
+      write(iunit,*) '  |                   ', COMPDEF_5
+      write(iunit,*) '  |                   ', COMPDEF_6
+      write(iunit,*) '  |                   ', COMPDEF_7
+      write(iunit,*) '  |                   ', COMPDEF_8
+      write(iunit,*) '  |                   ', COMPDEF_9
+      write(iunit,*) '  |                   ', COMPDEF_10
+      write(iunit,*) '  |                   ', COMPDEF_11
+      write(iunit,*) '  |                   ', COMPDEF_12
+      write(iunit,*) '  |                   ', COMPDEF_13
+      write(iunit,*) '  |                   ', COMPDEF_14
+      write(iunit,*) '  |                   ', COMPDEF_15
+      write(iunit,*) '  |                   ', COMPDEF_16
+      write(iunit,*) '  |                   ', COMPDEF_17
+      write(iunit,*) '  +-----------------------------------------------------------'
+    end subroutine print_build_info
+  end module mcnp_env
diff -rN '--unified=0' MCNP611/Source/src/msgtsk.F90 MCNP611/Source_new/src/msgtsk.F90
--- MCNP611/Source/src/msgtsk.F90
+++ MCNP611/Source_new/src/msgtsk.F90
@@ -80,0 +81 @@
+  use, intrinsic :: iso_c_binding
@@ -308 +309 @@
-    call setrngcgm(rang) 
+    call setrngcgm(c_funloc(rang))
diff -rN '--unified=0' MCNP611/Source/src/pblcom.F90 MCNP611/Source_new/src/pblcom.F90
--- MCNP611/Source/src/pblcom.F90
+++ MCNP611/Source_new/src/pblcom.F90
@@ -178,4 +178,6 @@
-  common /pblcom_thread/ pbl, pbl_src, pbl_stack
-  common /pblcom_thread/ udt, udt_stack
-  common /pblcom_thread/ pbl_in_use, udt_in_use, is_uncollided_part
-  !$OMP THREADPRIVATE( /pblcom_thread/ )
+  common /pblcom_thread1/ pbl, pbl_src, pbl_stack
+  common /pblcom_thread2/ udt, udt_stack
+  common /pblcom_thread3/ pbl_in_use, udt_in_use, is_uncollided_part
+  !$OMP THREADPRIVATE( /pblcom_thread1/ )
+  !$OMP THREADPRIVATE( /pblcom_thread2/ )
+  !$OMP THREADPRIVATE( /pblcom_thread3/ )
diff -rN '--unified=0' MCNP611/Source/src/torus.F90 MCNP611/Source_new/src/torus.F90
--- MCNP611/Source/src/torus.F90
+++ MCNP611/Source_new/src/torus.F90
@@ -86 +86,7 @@
-      if( abs(t3)<=z0 )  go to 40
+      if( abs(t3)<=z0 ) then
+        if (xx(m) < 0.) then
+          cycle DO_60
+        else
+          go to 40
+        endif
+      endif
