diff -rN '--unified=0' MCNP514/Source/dotcomm/include/dotcomm.h MCNP514/Source_new/dotcomm/include/dotcomm.h
--- MCNP514/Source/dotcomm/include/dotcomm.h
+++ MCNP514/Source_new/dotcomm/include/dotcomm.h
@@ -164 +164 @@
-#define DOTCOMM_Datatype   int
+#define DOTCOMM_Datatype   MPI_Datatype
@@ -584 +584 @@
-				 int             *datatype,
+				 DOTCOMM_Datatype *datatype,
diff -rN '--unified=0' MCNP514/Source/dotcomm/src/internals/mpi/dotcommi_enroll.c MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommi_enroll.c
--- MCNP514/Source/dotcomm/src/internals/mpi/dotcommi_enroll.c
+++ MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommi_enroll.c
@@ -60,0 +61,2 @@
+  DOTCOMM_Context  
+    default_user_context;   /* Set to MPI_COMM_WORLD.                        */
@@ -62 +63,0 @@
-    default_user_context,   /* Set to MPI_COMM_WORLD.                        */
@@ -93 +94 @@
-  DOTCOMMI_SET_CONTEXT ( &default_user_context, ierr );
+  DOTCOMMI_SET_CONTEXT ( (int*)&default_user_context, ierr );
diff -rN '--unified=0' MCNP514/Source/dotcomm/src/internals/mpi/dotcommi_pack.c MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommi_pack.c
--- MCNP514/Source/dotcomm/src/internals/mpi/dotcommi_pack.c
+++ MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommi_pack.c
@@ -60 +60 @@
-void DOTCOMMI_PACK ( const void *pv, int *count, int *datatype, int *ierr)
+void DOTCOMMI_PACK ( const void *pv, int *count, DOTCOMM_Datatype *datatype, int *ierr)
diff -rN '--unified=0' MCNP514/Source/dotcomm/src/internals/mpi/dotcommpi_mpi.h MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommpi_mpi.h
--- MCNP514/Source/dotcomm/src/internals/mpi/dotcommpi_mpi.h
+++ MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommpi_mpi.h
@@ -35 +35 @@
-  int dotcommpi_sizeof ( int datatype );
+  int dotcommpi_sizeof ( DOTCOMM_Datatype datatype );
diff -rN '--unified=0' MCNP514/Source/dotcomm/src/internals/mpi/dotcommpi_sizeof.c MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommpi_sizeof.c
--- MCNP514/Source/dotcomm/src/internals/mpi/dotcommpi_sizeof.c
+++ MCNP514/Source_new/dotcomm/src/internals/mpi/dotcommpi_sizeof.c
@@ -40 +40 @@
-int dotcommpi_sizeof ( int datatype )
+int dotcommpi_sizeof ( DOTCOMM_Datatype datatype )
diff -rN '--unified=0' MCNP514/Source/src/mc.c MCNP514/Source_new/src/mc.c
--- MCNP514/Source/src/mc.c
+++ MCNP514/Source_new/src/mc.c
@@ -437 +437 @@
-  extern void xevent();
+  extern void xevent(IDEF wait);
@@ -469 +469 @@
-  extern void xevent();
+  extern void xevent(IDEF wait);
@@ -528 +528 @@
-  extern void xevent();
+  extern void xevent(IDEF wait);
diff -rN '--unified=0' MCNP514/Source/src/torus.F90 MCNP514/Source_new/src/torus.F90
--- MCNP514/Source/src/torus.F90
+++ MCNP514/Source_new/src/torus.F90
@@ -68 +68,7 @@
-      if( abs(t3)<=z0 )  go to 40
+      if( abs(t3)<=z0 ) then
+        if (xx(m) < 0.) then
+          cycle DO_60
+        else
+          go to 40
+        endif
+      endif
